pallets/runtime/tests/src/settlement_test.rs

```diff
@@ -3060,3 +3076,437 @@ fn multiple_custodian_settlement() {
+                instruction_counter,
+                finalized_tx,
+                default_portfolio_vec(bob_did),
+            ));
+
+            // ------ Mediator authorizes.
+            // Mediator reads the receiver's proofs from the chain (it contains the sender's proofs as well).
+            println!("-------------> Charlie is going to authorize.");
+            let mut tx_data = Settlement::mercat_tx_data(instruction_counter);
+            assert_eq!(tx_data.len(), 2);
+
+            let tx_data = tx_data.remove(1);
+            let decoded_finalized_tx = match tx_data {
+                MercatTxData::FinalizedTransfer(finalized) => {
+                    let mut data: &[u8] = &finalized;
+                    FinalizedTransferTx::decode(&mut data).unwrap()
+                }
+                _ => {
+                    panic!("Unexpected data type");
+                }
+            };
+
+            // Mediator verifies the proofs in the wallet.
+            let justified_tx = MercatTxData::JustifiedTransfer(
+                CtxMediator {}
+                    .justify_transaction(
+                        decoded_finalized_tx,
+                        &charlie_secret_account.enc_keys,
+                        &alice_public_account,
+                        &alice_encrypted_init_balance,
+                        &bob_public_account,
+                        &[],
+                        AssetId {
+                            id: *ticker.as_bytes(),
+                        },
+                        &mut rng,
+                    )
+                    .unwrap()
+                    .encode(),
+            );
+
+            println!("-------------> This should trigger the execution");
+            assert_ok!(Settlement::authorize_confidential_instruction(
+                Origin::signed(charlie),
+                instruction_counter,
+                justified_tx,
+                default_portfolio_vec(charlie_did),
+            ));
+
+            // Instruction should've settled.
+            // Verify by decrypting the new balance of both Alice and Bob.
+            let new_alice_balance =
+                ConfidentialAsset::mercat_account_balance(alice_did, alice_account_id)
+                    .to_mercat::<TestStorage>()
+                    .unwrap();
+            let new_alice_balance = alice_secret_account
+                .enc_keys
+                .scrt
+                .decrypt(&new_alice_balance)
+                .unwrap();
+            assert_eq!(new_alice_balance as u128, total_supply - amount as u128);
+
+            let new_bob_balance =
+                ConfidentialAsset::mercat_account_balance(bob_did, bob_account_id)
+                    .to_mercat::<TestStorage>()
+                    .unwrap();
+            let new_bob_balance = bob_secret_account
+                .enc_keys
+                .scrt
+                .decrypt(&new_bob_balance)
+                .unwrap();
```

Comment:
Instead of decrypting you could compare with the expected CipherText.

Reply:
If we cannot run the tests faster, then I'll definitely do this. Otherwise, I think that decrypting provides a higher level of assurance on the correctness of the test.

Reply:
decrypting these doesn't add more assurance than comparing the ciphertext with ciphertext you expect. The only minor caveat is that if anyone touches this test to change the seed, they'll break the test, because that'd change the cipher text.